storm-client/src/jvm/org/apache/storm/coordination/BatchSubtopologyBuilder.java

```diff
@@ -84,24 +84,24 @@ private BoltDeclarer setBolt(String id, IRichBolt bolt, Number parallelism) {
             p = parallelism.intValue();
         }
         Component component = new Component(bolt, p);
-        _bolts.put(id, component);
+        bolts.put(id, component);
         return new BoltDeclarerImpl(component);
     }
     
     public void extendTopology(TopologyBuilder builder) {
-        BoltDeclarer declarer = builder.setBolt(_masterId, new CoordinatedBolt(_masterBolt.bolt), _masterBolt.parallelism);
-        for (InputDeclaration decl: _masterBolt.declarations) {
+        BoltDeclarer declarer = builder.setBolt(masterId, new CoordinatedBolt(masterBolt.bolt), masterBolt.parallelism);
+        for (InputDeclaration decl: masterBolt.declarations) {
             decl.declare(declarer);
         }
-        for (Map<String, Object> conf: _masterBolt.componentConfs) {
-            declarer.addConfigurations(conf);
+        if (!masterBolt.componentConf.isEmpty()) {
```

Comment:
Nit: addConfigurations seems like it does this check too, so it seems unnecessary here

Reply:
It is probably unnecessary and I will pull it out.  I left it in initially for compatibility, why try to add an empty config, which will be a no-op.